types/fusebit__oauth-connector/index.d.ts

```diff
@@ -0,0 +1,162 @@
+// Type definitions for @fusebit/oauth-connector 1.3
+// Project: http://fusebit.io/
+// Definitions by: Andrew Miller <https://github.com/andrewrmiller>
+//                 Trent Hashimoto <https://github.com/hashiphoto>
+//                 Andy Dam <https://github.com/andydam>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+
+import { FusebitContext } from '@fusebit/add-on-sdk';
+import * as express from 'express';
+
+/**
+ * Model for persisted user data.
+ */
+export interface UserContext<TVendorUserProfile = object> {
+    status?: string;
+    timestamp: number;
+    vendorUserId: string;
+    vendorUserProfile: TVendorUserProfile;
+    vendorToken?: {
+        scope: string;
+        expires_at: number;
+        expires_in: number;
+        token_type: string;
+        access_token: string;
+        refresh_token: string;
+        ext_expires_in: number;
+    };
+    foreignOAuthIdentities: {
+        [key: string]: {
+        userId: string;
+        connectorBaseUrl: string;
+        };
+    };
+}
+
+/**
+ * Parameters that are passed to the authorize method.
+ */
+export interface AuthorizeParams {
+    action: string;
+    resourceFactory: (req: express.Request) => string;
+}
+
+/**
+ * Response to a request for an access token.
+ */
+export interface OAuthTokenResponse {
+    access_token: string;
+    token_type: string;
+    expires_in: string;
+    refresh_token?: string;
+    scope?: string;
+}
+
+export class OAuthConnector {
+    /**
+     * Called during connector initialization to allow the connector to register additional, application-specific
+     * routes on the provided Express router.
+     *
+     * @param Express router
+     */
+    onCreate(app: express.Router): void;
+
+    /**
+     * Creates Express middleware that authorizes the call using Fusebit security. For example, the following will only execute
+     * the Express handler if the access token supplied by the caller has the function:execute permission on the function resource.
+     *
+     * app.get('/myendpoint',
+     *   authorize({
+     *     action: 'function:execute',
+     *     resourceFactory: req => `/account/${req.fusebit.accountId}/subscription/${req.fusebit.subscriptionId}/boundary/${req.fusebit.boundaryId}/function/${req.fusebit.functionId}/myendpoint/`
+     *   }),
+     *   handler
+     * );
+     *
+     * @param param Object with action and resourceFactory properties
+     */
+    authorize(params: AuthorizeParams): express.RequestHandler;
+
+    /**
+     * Called after a new user successfuly completed a configuration flow and was persisted in the system. This extensibility
+     * point allows for creation of any artifacts required to serve this new user, for example creation of additional
+     * Fusebit functions.
+     *
```

Comment:
```suggestion
     *
     * @async
     * 
```

Reply:
this is optional, one can use `@async` to denote methods defined with `async` keyword:
https://jsdoc.app/tags-async.html

Reply:
OK.  Since we're still waiting on reviews I'll go ahead and add this.